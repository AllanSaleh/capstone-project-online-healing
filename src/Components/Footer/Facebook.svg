<svg width="35" height="35" viewBox="0 0 35 35" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M1.45825 17.539C1.45825 8.65794 8.64035 1.45837 17.4999 1.45837C26.3595 1.45837 33.5416 8.65794 33.5416 17.539C33.5416 25.8822 27.2031 32.7414 19.09 33.5417C19.1335 33.5341 19.1768 33.5259 19.2201 33.5172V22.1551H23.3906L24.0147 17.3013H19.2201V14.2024C19.2201 12.797 19.6092 11.8396 21.6198 11.8396L24.1836 11.8383V7.49719C23.7401 7.43784 22.2183 7.30589 20.4475 7.30589C16.751 7.30589 14.2198 9.56768 14.2198 13.7219V17.3013H10.0386V22.1551H14.2198V33.2833C6.93343 31.7657 1.45825 25.2933 1.45825 17.539Z" fill="#1A1A1A"/>
</svg>
