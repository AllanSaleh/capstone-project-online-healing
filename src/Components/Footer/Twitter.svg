<svg width="35" height="35" viewBox="0 0 35 35" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M29.2578 4.62682C30.9347 4.28241 32.5884 3.39473 32.5884 3.39473C33.3256 3.04856 33.6012 3.31906 33.2037 4.01651C33.2037 4.01651 32.2112 6.07413 30.6949 7.02984C32.1797 6.84341 33.52 6.29924 33.52 6.29924C34.2926 6.0365 34.5007 6.66055 33.9952 7.29167C33.9952 7.29167 32.6842 9.11819 31.25 10.2083C31.2624 10.538 31.2682 10.4798 31.2682 10.8164C31.2682 22.4922 25.3761 32.8125 10.3789 32.8125C4.74279 32.8125 1.72544 30.5762 1.72544 30.5762C0.269777 29.8354 0.435023 28.9706 2.05229 28.7287C2.05229 28.7287 7.48179 28.2115 9.99381 26.1372C-4.37503 18.9583 1.48563 5.05932 1.48563 5.05932C1.54196 4.25827 2.09591 4.05424 2.68917 4.61642C2.68917 4.61642 9.15627 11.3561 16.7492 11.6789C16.6231 11.1127 16.5601 10.525 16.5601 9.91857C16.5601 5.64926 19.8479 2.1875 23.8993 2.1875C26.0129 2.1875 27.922 3.12555 29.2578 4.62682Z" fill="#1A1A1A"/>
</svg>
