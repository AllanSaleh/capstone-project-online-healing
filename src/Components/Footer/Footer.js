import React from 'react';
import Send from './Send.svg';
import Twitter from './Twitter.svg';
import Facebook from './Facebook.svg';
import Google from './Google.svg';

export default function Footer() {
  return (
    <div className="flex justify-between px-sides items-center h-footer bg-yellow-primary md:text-sm">
      <div className="flex flex-col justify-between items-start h-40">
        <div>
          <div className="text-title leading-tight">Subscribe</div>
          <div className="text-paragraph text-gray-700">
            We'll never try to spam you or share your e-mail
          </div>
        </div>

        <div className="flex border-gray-700 border-2 h-12 rounded-lg">
          <input
            className="w-64 p-2 rounded-l-md"
            placeholder="Enter your E-Mail"
            type="email"
          />
          <button
            className="border-0 border-l-2 border-gray-700 p-2 bg-blue-dark rounded-r-md"
            type="button"
          >
            <img src={Send} alt="Send" />
          </button>
        </div>
      </div>

      <div className="flex flex-col justify-evenly items-center h-40">
        <div className="flex justify-between items-center text-gray-700 text-paragraph">
          <a
            onClick={() => window.scrollTo(0, 0)}
            href="#"
            className="mx-4 hover:underline hover:text-blue-dark"
          >
            Home
          </a>
          <a
            onClick={() => window.scrollTo(0, 0)}
            href="#"
            className="mx-4 hover:underline hover:text-blue-dark"
          >
            Blogs
          </a>
          <a
            onClick={() => window.scrollTo(0, 0)}
            href="#"
            className="mx-4 hover:underline hover:text-blue-dark"
          >
            About
          </a>
          <a
            onClick={() => window.scrollTo(0, 0)}
            href="#"
            className="mx-4 hover:underline hover:text-blue-dark"
          >
            Contact Us
          </a>
        </div>

        <div className="flex justify-center items-center">
          <a href="#" target="_blank" className="mx-4">
            <img src={Twitter} alt="Twitter" />
          </a>
          <a href="#" target="_blank" className="mx-4">
            <img src={Facebook} alt="Facebook" />
          </a>
          <a href="#" target="_blank" className="mx-4">
            <img src={Google} alt="Google" />
          </a>
        </div>
      </div>
    </div>
  );
}
