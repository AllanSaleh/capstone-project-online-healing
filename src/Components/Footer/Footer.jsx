import React, { useState } from 'react';
import { Link, useHistory } from 'react-router-dom';
import { useTranslation } from 'react-i18next'

import Send from './Images/Send.svg';
import Twitter from './Images/Twitter.svg';
import Facebook from './Images/Facebook.svg';
import Google from './Images/Google.svg';

export default function Footer() {
  const history = useHistory();

  const [email, setEmail] = useState('');

  const subscribe = () => {
    if (
      email.match(
        /[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9]+[.]+[a-zA-Z0-9]/
      )
    )
      history.push({
        pathname: '/ThankYou',
        state: 'You email has been added to the mailing list successfully!',
      });
    else alert('Please enter a valid E-Mail!');
  };

  const { t, i18n } = useTranslation();
  function changeLang(lang) {
    i18n.changeLanguage(lang);
  }

  return (
    <div className="py-4 flex flex-col lg:flex-row justify-between px-sides items-center lg:h-footer bg-yellow-primary">
      <div className="flex flex-col justify-between items-center text-center lg:text-left lg:items-start h-40">
        <div>
          <div className="text-3xl lg:text-title">{t('Footer.Subscribe')}</div>
          <div className="text-xl lg:text-paragraph lg:mt-4 text-gray-700">{t('Footer.Spam')}</div>
        </div>

        <div className="flex border-gray-700 border-2 h-12 rounded-lg">
          <input
            onChange={(e) => setEmail(e.target.value)}
            className="w-64 p-2 rounded-l-md"
            placeholder={t('Footer.Enter')}
            type="email"
          />
          <button
            onClick={() => subscribe()}
            className="border-0 border-l-2 border-gray-600 p-2 bg-blue-dark rounded-r-md"
            type="button"
          >
            <img src={Send} alt="Send" />
          </button>
        </div>
      </div>

      <div className="flex flex-col justify-evenly items-center h-40">
        <div className="w-72 lg:w-96 flex justify-between items-center text-gray-700 text-lg lg:text-paragraph">
          <Link
            to="/"
            onClick={() => window.scrollTo(0, 0)}
            className="hover:underline hover:text-blue-dark"
          >{t('Home')}</Link>
          <Link
            to="/Blog"
            onClick={() => window.scrollTo(0, 0)}
            className="hover:underline hover:text-blue-dark"
          >{t('Blogs')}</Link>
          <Link
            to="/About"
            onClick={() => window.scrollTo(0, 0)}
            className="hover:underline hover:text-blue-dark"
          >{t('About')}</Link>
          <Link
            to="/Contact"
            onClick={() => window.scrollTo(0, 0)}
            className="hover:underline hover:text-blue-dark"
          >{t('Contact')}</Link>
        </div>

        <div className="flex justify-center items-center">
          <a href="https://twitter.com/" target="_blank" className="mx-4">
            <img src={Twitter} alt="Twitter" />
          </a>
          <a href="https://www.facebook.com/" target="_blank" className="mx-4">
            <img src={Facebook} alt="Facebook" />
          </a>
          <a href="https://www.google.com/" target="_blank" className="mx-4">
            <img src={Google} alt="Google" />
          </a>
        </div>

        <div className="flex justify-center items-center">
          <label className="w-24">{t('Footer.Language')}:</label>
          <select
            onChange={(e) => changeLang(e.target.value)}
            className="w-24 p-1 border-gray-700 border-2 rounded-lg"
          >
            <option value="en" className="py-1 text-paragraph text-gray-700">
              English
            </option>
            <option value="tr" className="py-1 text-paragraph text-gray-700">
              Türkçe
            </option>
          </select>
        </div>
      </div>
    </div>
  );
}
