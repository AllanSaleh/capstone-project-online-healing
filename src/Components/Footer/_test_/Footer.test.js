import * as React from 'react';
import Footer from "../Footer";
import * as renderer from "react-test-renderer";

it('Footer renders correctly', () => {
  const tree = renderer
    .create(<Footer/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});