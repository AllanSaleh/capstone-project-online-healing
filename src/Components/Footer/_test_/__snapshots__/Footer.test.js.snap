// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<div
  className="py-4 flex flex-col lg:flex-row justify-between px-sides items-center lg:h-footer bg-yellow-primary"
>
  <div
    className="flex flex-col justify-between items-center text-center lg:text-left lg:items-start h-40"
  >
    <div>
      <div
        className="text-3xl lg:text-title"
      >
        Subscribe
      </div>
      <div
        className="text-xl lg:text-paragraph lg:mt-4 text-gray-700"
      >
        We'll never try to spam you or share your e-mail
      </div>
    </div>
    <div
      className="flex border-gray-700 border-2 h-12 rounded-lg"
    >
      <input
        className="w-64 p-2 rounded-l-md"
        placeholder="Enter your E-Mail"
        type="email"
      />
      <button
        className="w-12 border-0 border-l-2 border-gray-700 p-2 bg-blue-dark rounded-r-md"
        type="button"
      >
        <img
          alt="Send"
          src="Send.svg"
        />
      </button>
    </div>
  </div>
  <div
    className="flex flex-col justify-evenly items-center h-40"
  >
    <div
      className="w-72 lg:w-96 flex justify-between items-center text-gray-700 text-lg lg:text-paragraph"
    >
      <a
        className="hover:underline hover:text-blue-dark"
        href="#"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        className="hover:underline hover:text-blue-dark"
        href="#"
        onClick={[Function]}
      >
        Blogs
      </a>
      <a
        className="hover:underline hover:text-blue-dark"
        href="#"
        onClick={[Function]}
      >
        About
      </a>
      <a
        className="hover:underline hover:text-blue-dark"
        href="#"
        onClick={[Function]}
      >
        Contact Us
      </a>
    </div>
    <div
      className="flex justify-center items-center"
    >
      <a
        className="mx-4"
        href="#"
        target="_blank"
      >
        <img
          alt="Twitter"
          src="Twitter.svg"
        />
      </a>
      <a
        className="mx-4"
        href="#"
        target="_blank"
      >
        <img
          alt="Facebook"
          src="Facebook.svg"
        />
      </a>
      <a
        className="mx-4"
        href="#"
        target="_blank"
      >
        <img
          alt="Google"
          src="Google.svg"
        />
      </a>
    </div>
    <div
      className="flex justify-center items-center"
    >
      <label
        className="w-24"
      >
        Language:
      </label>
      <select
        className="w-24 p-1 border-gray-700 border-2 rounded-lg"
      >
        <option
          className="py-1 text-paragraph text-gray-700"
          value="En"
        >
          English
        </option>
        <option
          className="py-1 text-paragraph text-gray-700"
          value="Ku"
        >
          كوردى
        </option>
      </select>
    </div>
  </div>
</div>
`;
